crates/nu-protocol/src/ty.rs

```diff
@@ -40,11 +40,11 @@ impl Type {
         let is_subtype_collection = |this: &[(String, Type)], that: &[(String, Type)]| {
             if this.is_empty() || that.is_empty() {
                 true
-            } else if this.len() > that.len() {
+            } else if this.len() < that.len() {
                 false
             } else {
-                this.iter().all(|(col_x, ty_x)| {
-                    if let Some((_, ty_y)) = that.iter().find(|(col_y, _)| col_x == col_y) {
+                that.iter().all(|(col_y, ty_y)| {
+                    if let Some((_, ty_x)) = this.iter().find(|(col_x, _)| col_x == col_y) {
```

Comment:
What is the purpose of changing the `is_subtype()` here?

Reply:
Because the original implementation seems wrong.  Before this change, it failed on the following case:
```Record<a: any, b: any>.is_subtype(Record<a: any>)```.  Which should return `true`

Reply:
Sure!  Updated